src/common/AndroidLauncher.ts

```diff
@@ -53,18 +54,18 @@ export class AndroidLauncher {
             messages.getMessage('startPreviewAction'),
             messages.getMessage('searchForDeviceStatus', [emuName])
         );
-        return AndroidUtils.hasEmulator(emuName)
+        return AndroidUtils.hasEmulator(emuName, logger)
             .then((result) => {
                 if (!result) {
                     CommonUtils.updateCliAction(messages.getMessage('createDeviceStatus', [emuName]));
-                    return AndroidUtils.createNewVirtualDevice(emuName, emuImage, androidApi, device, abi);
+                    return AndroidUtils.createNewVirtualDevice(emuName, emuImage, androidApi, device, abi, logger);
                 }
                 CommonUtils.updateCliAction(messages.getMessage('foundDeviceStatus', [emuName]));
                 return Promise.resolve();
             })
             .then(() => {
                 CommonUtils.updateCliAction(messages.getMessage('startDeviceStatus', [emuName]));
-                return AndroidUtils.startEmulator(emuName);
+                return AndroidUtils.startEmulator(emuName, false, true, logger);
```

Comment:
I with typescript would support calling functions using named parameter rather than me just passing `true` and `false`